ext/limiter/limiter.c

```diff
@@ -0,0 +1,158 @@
+#include "limiter.h"
+
+#include <pthread.h>
+#include <unistd.h>
+#include <sys/mman.h>
+
+// clang-format Off
+
+#define NANOSECONDS_PER_SECOND 1000000000
+
+typedef struct {
+    /* limit from configuration DD_TRACE_RATE_LIMIT */
+    uint32_t limit;
+    struct {
+        /* beginning of window */
+        uint64_t open;
+        /* last hit inside this window */
+        uint64_t current;
+        /* total count for this window */
+        uint32_t samples;
+        /* allowed count for this window */
+        uint32_t allowed;
+        /* limit (tokens) for this window */
+        uint32_t limit;
+        /* rate for the last window that passed */
+        uint32_t rate;
+    } window;
+
+    pthread_mutex_t lock;
+} ddtrace_limiter;
+
+static ddtrace_limiter* __limiter;
```

Comment:
Can you please rename it to `dd_limiter` to conform with our naming schema for static vars? :-)